.github/workflows/test.yaml

```diff
@@ -513,16 +513,47 @@ jobs:
       run: |
         ls -R dist/
 
+    - uses: actions/upload-artifact@v4
+      with:
+        name: dist
+        path: dist
+
     - name: Check matched tag version and branch version - on tag
       if: startsWith(github.ref, 'refs/tags/twisted-')
       run: python admin/check_tag_version_match.py "${{ github.ref }}"
 
-    - name: Publish to PyPI - on tag
-      if: startsWith(github.ref, 'refs/tags/twisted-')
-      uses: pypa/gh-action-pypi-publish@v1.5.1
+  # Used exclusively for various release publishing.
+  # This is only executed on tags, but just does the publish to exercise the release as much
+  # as possible and reduce the possibility of finding bugs in the release
+  # process late in the release cycle,
+  release-publish:
+    needs:
+      release-build
+    if: startsWith(github.ref, 'refs/tags/twisted-')
+    environment:
+      release
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    name: publish on twisted-* tag
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/download-artifact@v4
       with:
-        password: ${{ secrets.PYPI_UPLOAD_TOKEN }}
+        name: dist
+        path: dist
+
+    - name: Publish to PyPI - on tag
+      uses: pypa/gh-action-pypi-publish@v1.12.2
 
+  release-publish-success-or-skipped:
```

Comment:
I hope that we can have this implemented without this extra job.

If we need to keep this job, I think that this needs a comment to explai why we need it.